### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by hirosassa**
Updated dependencies for following reasons:
- To follow newer go version
- To use workload identity federation by using newer version of `golang.org/x/oauth2`
  - It needs [this line](https://cs.opensource.google/go/x/oauth2/+/master:google/google.go;l=97).

Using current latest berglas provider with a credential obtained from workload identity federation, we saw following error. I would like to resolve it.

> Error: failed to configure provider: failed to parse credentials: unknown credential type: "external_account"
>  with provider["registry.terraform.io/sethvargo/berglas"]

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,43 +14,77 @@
...
-go 1.16
+go 1.19
 
 require (
-	cloud.google.com/go v0.93.3 // indirect
-	cloud.google.com/go/kms v0.1.0 // indirect
-	cloud.google.com/go/secretmanager v0.1.0 // indirect
-	cloud.google.com/go/storage v1.16.0 // indirect
-	github.com/GoogleCloudPlatform/berglas v0.6.0
+	github.com/GoogleCloudPlatform/berglas v0.6.3-0.20220726215926-83e4b9214bea
```
Comment (hirosassa) [inline go.mod:20]: I could not build this provider by using current latest tagged version of berglas client due to following error:

> ../../go/pkg/mod/github.com/!google!cloud!platform/berglas@v0.6.1/pkg/berglas/iam_storage.go:201:12: assignment mismatch: 2 variables but retry.NewFibonacci returns 1 value

This error comes from [this](https://github.com/GoogleCloudPlatform/berglas/blob/main/pkg/berglas/iam_storage.go#L199-L201) change, so I currently refer latest commit of the client.

**Comment by hirosassa** [comment]
@sethvargo Thank you for the quick review and merge!

**Comment by hirosassa** [comment]
@sethvargo v0.2.1 is not released to terraform registry yet?
https://registry.terraform.io/providers/sethvargo/berglas/0.2.0

**Comment by sethvargo** [comment]
@hirosassa so - I've manually refreshed it at least 3x now...

screenshot-20220815-OYxnBteM@2x


I think the problem might be that I have ECC GPG signing keys and it looks like the Terraform registry [doesn't support ECC keys](https://www.terraform.io/registry/providers/publishing#preparing-and-adding-a-signing-key). I don't have RSA/DSA keys and don't have intentions of managing multiple GPG keys right now.

**Comment by hirosassa** [comment]
@sethvargo Thanks for switching back [commit](https://github.com/sethvargo/terraform-provider-berglas/commit/4cec79cc07768fcc130b8ac3b6f3455ca0da8d80).
So can we re-run release job (by adding new tag?) to release now?

**Comment by sethvargo** [comment]
That won't fix it - the Terraform registry doesn't support the signature I'm using.

**Pull request merged**